server/src/main/java/org/elasticsearch/common/geo/parsers/ShapeParser.java

```diff
@@ -52,8 +52,6 @@ static ShapeBuilder parse(XContentParser parser) throws IOException {
         }
         if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
             return GeoJsonParser.parse(parser);
-        } else if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
```

Comment:
The main entry point to ShapeParser.parse is from `GeoJSONUtils.geoJSONString2ShapeBuilder` - which cannot contain a WKT string.

To parse WKT strings we use `GeoPointType.WKT_READER.parse(wkt)` elsewhere.